src/lib.rs

```diff
@@ -79,6 +79,49 @@ fn maybe_remove_convert_frame_suffixes(frames: &mut Vec<FrameSummary>) {
     }
 }
 
+fn add_unique_suffix(raw_filename: PathBuf, output_count: i32) -> PathBuf {
+    if let Some(stem) = raw_filename.file_stem() {
+        let mut r = OsString::new();
+        r.push(stem);
+        r.push(OsStr::new("_"));
+        r.push(output_count.to_string());
+        if let Some(e) = raw_filename.extension() {
+            r.push(OsStr::new("."));
+            r.push(e);
+        };
+        raw_filename.with_file_name(r)
+    } else {
+        raw_filename
+    }
+}
+
+fn add_file_output(
+    filename: PathBuf,
+    content: String,
+    output: &mut Vec<(PathBuf, String)>,
+    compile_directory: &mut Vec<OutputFile>,
+    output_count: &mut i32,
+) {
+    output.push((filename.clone(), content));
+    let filename_str = format!("{}", filename.to_string_lossy());
```

Comment:
nit you could just call `filename.to_string_lossy().to_string()`  or `filename.to_string_lossy().into_owned()`